src/services/user.service.js

```diff
@@ -80,11 +81,24 @@ const deleteUserById = async (userId) => {
   return user;
 };
 
+const getStudents = async (userId) => {
+  const classes = await Class.find({ teacher: userId }).populate('students');
+  if (!classes) {
+    throw new ApiError(httpStatus.NOT_FOUND, 'Classes not found');
+  }
+  const students = new Set();
+  logger.info(students);
+  for (let i = 0; i < classes.length; i += 1)
+    for (let j = 0; j < classes[i].students.length; j += 1) students.add(classes[i].students[j]);
```

Comment:
```suggestion
  classes.forEach((class) => class.students.forEach((student) => students.add(student)))
```
this would be much cleaner